### PR TITLE
Add installation of diffutils to fix Ventura when running the go linter

### PR DESCRIPTION
**Pull request by YanZ777**
## Summary

Folks were seeing issues with the Go linter when committing. 
diffutils needed to be installed, this PR adds it to the Brew file. 

I updated the nix/default.nix file using the instructions from [this PR](https://github.com/transcom/mymove/pull/9622#pullrequestreview-1186864800).

## What I'm looking for to be reviewed
That the code I wrote makes sense.
That the code works. 

This is the error that people were seeing: 
```
* can't run linter goanalysis_metalinter: gofmt: can't extract issues from gofmt diff output
```

[This is the message in Slack](https://ustcdp3.slack.com/archives/CP6PTUPQF/p1666880730667879) that points out this issue.

Full error output
``` 
* can't run linter goanalysis_metalinter: gofmt: can't extract issues from gofmt diff output "--- /Users/sampleuser/code/mymove/pkg/assets/assets.go.orig\t2022-10-27 14:12:12\n+++ /Users/sampleuser/code/mymove/pkg/assets/assets.go\t2022-10-27 14:12:12\n@@ -630,11 +630,13 @@\n // directory embedded in the file by go-bindata.\n // For example if you run go-bindata on data/... and data contains the\n // following hierarchy:\n-//     data/\n-//       foo.txt\n-//       img/\n-//         a.png\n-//         b.png\n+//\n+//\tdata/\n+//\t  foo.txt\n+//\t  img/\n+//\t    a.png\n+//\t    b.png\n+//\n // then AssetDir(\"data\") would return []string{\"foo.txt\", \"img\"},\n // AssetDir(\"data/img\") would return []string{\"a.png\", \"b.png\"},\n // AssetDir(\"foo.txt\") and AssetDir(\"notexist\") would return an error, and\n": can't parse patch: parsing time "2022-10-27 14:12:12" as "2006-01-02 15:04:05 -0700": cannot parse "" as "-0700"
```

### Additional steps

1. Install Ventura.
2. Make some changes to the Go code. 
3. Try to commit it. 
4. The commit should and not fail.

### CODE REVIEW

File: nix/default.nix [hunk 1]
```diff
@@ -200,6 +200,13 @@ buildEnv {
       })
       { }).postgresql_jdbc
 
+    (import (builtins.fetchGit {
+          # Descriptive name to make the store path easier to identify
+          name = "diffutils-3.8";
+          url = "https://github.com/NixOS/nixpkgs/";
+          ref = "refs/heads/nixpkgs-unstable";
+          rev = "bd4bfd814e96488febb38b4b4d0549c56b840c7f";
+        }) {}).diffutils
```
Comment (ronaktruss) [inline nix/default.nix:209]: Shouldn't we follow the existing pattern of having import and the builtins on different lines? Or is that the default based on line lengths? Similarly for having `{ }).diffutils` on a different line.

Comment (YanZ777) [reply nix/default.nix]: I blame copy paste! I should have checked more carefully though, thanks for spotting this. 
Do you know if there is a formatter for that file?

**Review by ronaktruss — APPROVED** [review]
Tested using nix and everything LGTM!

**Review by felipe-lee — APPROVED** [review]
So, I had installed `diffutils` with `brew` previously when I updated to ventura and uninstalled it prior to trying out this branch. When I ran `pre-commit run -a golangci-lint` without `diffutils`, I didn't get any errors, so maybe the issue has been fixed?

I will say though that this PR works for me, and seems to install the expected version.

**Comment by felipe-lee** [comment]
Ah, yeah, looks like it was fixed, so maybe this isn't needed? https://github.com/golangci/golangci-lint/issues/3087#issuecomment-1305983199

**Comment by YanZ777** [comment]
> Ah, yeah, looks like it was fixed, so maybe this isn't needed? [golangci/golangci-lint#3087 (comment)](https://github.com/golangci/golangci-lint/issues/3087#issuecomment-1305983199)

I went with merging this PR anyways since I am worried about someone being in some intermediary state.

**Comment by felipe-lee** [comment]
> > Ah, yeah, looks like it was fixed, so maybe this isn't needed? [golangci/golangci-lint#3087 (comment)](https://github.com/golangci/golangci-lint/issues/3087#issuecomment-1305983199)
> 
> I went with merging this PR anyways since I am worried about someone being in some intermediary state.

That is probably the better plan...I just ran into this on a different branch:

<details>
<summary>golangci-lint error</summary>
> golangci-lint............................................................Failed
> - hook id: golangci-lint
> - exit code: 3
level=warning msg="[runner] Can't run linter goanalysis_metalinter: goimports: can't extract issues from gofmt diff output \"--- /Users/felipe/dev/milmove/mymove/pkg/services/mocks/AdminUserCreator.go.orig\\t2022-12-01 00:33:41\\n+++ /Users/felipe/dev/milmove/mymove/pkg/services/mocks/AdminUserCreator.go\\t2022-12-01 00:33:41\\n@@ -4,6 +4,7 @@\\n \\n import (\\n \\tmock \\\"github.com/stretchr/testify/mock\\\"\\n+\\n \\tappcontext \\\"github.com/transcom/mymove/pkg/appcontext\\\"\\n \\n \\tmodels \\\"github.com/transcom/mymove/pkg/models\\\"\\n\": can't parse patch: parsing time \"2022-12-01 00:33:41\" as \"2006-01-02 15:04:05 -0700\": cannot parse \"\" as \"-0700\""
level=error msg="Running error: 1 error occurred:\n\t* can't run linter goanalysis_metalinter: goimports: can't extract issues from gofmt diff output \"--- /Users/felipe/dev/milmove/mymove/pkg/services/mocks/AdminUserCreator.go.orig\\t2022-12-01 00:33:41\\n+++ /Users/felipe/dev/milmove/mymove/pkg/services/mocks/AdminUserCreator.go\\t2022-12-01 00:33:41\\n@@ -4,6 +4,7 @@\\n \\n import (\\n \\tmock \\\"github.com/stretchr/testify/mock\\\"\\n+\\n \\tappcontext \\\"github.com/transcom/mymove/pkg/appcontext\\\"\\n \\n \\tmodels \\\"github.com/transcom/mymove/pkg/models\\\"\\n\": can't parse patch: parsing time \"2022-12-01 00:33:41\" as \"2006-01-02 15:04:05 -0700\": cannot parse \"\" as \"-0700\"\n\n"
</details>

I hadn't merged main in yet and still had `diffutils` uninstalled, so I guess maybe it's sometimes an issue still. I'm glad you've fixed it!

**Pull request merged**